src/Utility/Text.php

```diff
@@ -590,6 +590,11 @@ public static function truncate($text, $length = 100, array $options = [])
             }
             $truncate = mb_substr($truncate, 0, $spacepos);
         }
+        // If truncate still empty, then we don't need to count ellipsis in the cut.
+        if(empty($truncate)){
```

Comment:
You are right, I did not think about this scenario, I will modify unit tests as well.